packet_test.go

```diff
@@ -45,12 +45,12 @@ func TestParsePacket(t *testing.T) {
 	buf := &bytes.Buffer{}
 	w := astikit.NewBitsWriter(astikit.BitsWriterOptions{Writer: buf})
 	w.Write(uint16(1)) // Invalid sync byte
-	_, err := parsePacket(astikit.NewBytesIterator(buf.Bytes()))
+	_, err := parsePacket(astikit.NewBytesIterator(buf.Bytes()), nil)
 	assert.EqualError(t, err, ErrPacketMustStartWithASyncByte.Error())
 
 	// Valid
 	b, ep := packet(packetHeader, *packetAdaptationField, []byte("payload"), true)
-	p, err := parsePacket(astikit.NewBytesIterator(b))
+	p, err := parsePacket(astikit.NewBytesIterator(b), nil)
 	assert.NoError(t, err)
 	assert.Equal(t, p, ep)
 }
```

Comment:
Could you add a test for the filter here?

Something like:

```go
_, err = parsePacket(astikit.NewBytesIterator(b), func(p *Packet) bool { return true })
assert.EqualError(t, err, ...)
```

Reply:
Done. Also added test case for TestDemuxerNew.

Reply:
👍